src/client.spec.ts

```diff
@@ -251,19 +251,35 @@ describe('ReactSDKClient', () => {
       expect(onUserUpdateListener).toBeCalledTimes(1);
     });
 
-    it('calls fetchqualifiedsegements internally on each setuser call', async () => {
+    it('does not call fetchqualifiedsegements on setUser if onready is not calleed initially', async () => {
       const instance = createInstance(config);
       jest.spyOn(instance, 'fetchQualifiedSegments').mockImplementation(async () => true);
 
       await instance.setUser({
         id: 'xxfueaojfe8&86',
       });
 
+      expect(instance.fetchQualifiedSegments).toBeCalledTimes(0);
+    });
+
+    it('calls fetchqualifiedsegements internally on each setuser call after onready', async () => {
+      const instance = createInstance(config);
+      jest.spyOn(instance, 'fetchQualifiedSegments').mockImplementation(async () => true);
+
+      await instance.setUser({
+        id: 'xxfueaojfe8&86',
+      });
+      await instance.onReady()
+
+      await instance.setUser({
+        id: 'xxfueaojfe8&87',
+      });
+
       await instance.setUser({
         id: 'xxfueaojfe8&87',
       });
 
-      expect(instance.fetchQualifiedSegments).toBeCalledTimes(2);
+      expect(instance.fetchQualifiedSegments).toBeCalledTimes(3);
```

Comment:
Should this only be 2 since `setUser` is called 2x after the `onReady()`? ...or maybe I'm confused about the wording of the test name. 😕

Reply:
Got it.  This is where a `// comment` may be a useful addition for the future.  

Thanks for explaining.